### DISCUSSION TITLE
Arr::of() & Arrayable instance

### DISCUSSION THREAD

**Discussion by amaelftah** [laravel/framework#40898] [Ideas] [answered]
since we have [Stringeable](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Support/Stringable.php) and we can do

```php

Str::of('hello')

```

I thought it would be convenient to have 
```php
 Arr::of($array) 
 
 //so instead of 
 Arr::first($names, fn($name) => $name == 'ahmed');
 
 //it will be
 Arr::of($names)->first(fn ($name) => $name === 'ahmed');
 ``` 
 
 the motivation behind this is I tried to add **arr()** helper method same as **str()** but couldn't do it cause we don't have Arrayable class like Stringable

**Comment by ralphjsmit** [comment, marked as answer]
Can’t you just use a collection for that?

```php
collect($names)->first(fn ($name) => $name === ‘ahmed’);

collect($names)->firstWhere(‘name’, ‘ahmed’);
```

**Reply by amaelftah** [reply]
yup collection will do it ... am just giving an example of $names array ... as I wanted to showcase something simple with this idea

**Reply by ralphjsmit** [reply]
I know, just wanted to say that we already have an 'Arrayable', but that it's called a collection. I also thought about this concept myself, but then I realized that Collection is in fact purely an arrayable.

**Reply by amaelftah** [reply]
aha got your point :+1:

**Reply by shaedrich** [reply]
That would only make sense if Collection was more specific&mdash;a bit like the Eloquent collections are an 'array' of objects, ideally from the same class.

Might be related to #48181
